gef.py

```diff
@@ -4757,6 +4761,62 @@ def __set_repeat_count(self, argv: List[str], from_tty: bool) -> None:
         return
 
 
+@register
+class PieCommand(GenericCommand):
+    """Manage the current loaded architecture"""
+
+    _cmdline_ = "arch"
+    _syntax_ = f"{_cmdline_} (list|get|set) ..."
+    _example_ = f"{_cmdline_}"
+
+    def __init__(self) -> None:
+        super().__init__(prefix=True)
+        return
+
+    def do_invoke(self, argv: List[str]) -> None:
+        if not argv:
+            self.usage()
+        return
+
+@register
+class ArchGetCommand(GenericCommand):
+    """Get the current loaded architecture"""
+
+    _cmdline_ = "arch get"
+    _syntax_ = f"{_cmdline_}"
+    _example_ = f"{_cmdline_}"
+
+    def do_invoke(self, args: List[str]) -> None:
+        gef_print(f"{Color.greenify('Arch')}: {gef.arch}")
+        gef_print(f"{Color.greenify('Reason')}: {gef.arch_reason}")
+
+
+@register
+class ArchSetCommand(GenericCommand):
+    """Set the current loaded architecture"""
+
+    _cmdline_ = "arch set"
+    _syntax_ = f"{_cmdline_} <arch>"
+    _example_ = f"{_cmdline_}"
+
+    def do_invoke(self, args: List[str]) -> None:
+        reset_architecture(args[0] if args else None)
+
+@register
+class ArchListCommand(GenericCommand):
+    """List the available architectures"""
+
+    _cmdline_ = "arch list"
+    _syntax_ = f"{_cmdline_}"
+    _example_ = f"{_cmdline_}"
+
+    def do_invoke(self, args: List[str]) -> None:
+        gef_print(Color.greenify("Available architectures:"))
+        for arch in __registered_architectures__.keys():
+            if type(arch) == str:
```

Comment:
Why would the registered architectures not be keyed by strings?

Reply:
Because of Abi objects that can't be used as keys from the `arch set` command